consultation_analyser/consultations/upload_consultation.py

```diff
@@ -0,0 +1,50 @@
+import json
+
+from django.utils.text import slugify
+
+from consultation_analyser.consultations.public_schema import ConsultationWithResponses
+
+from .models import Answer, Consultation, ConsultationResponse, Question, Section
+
+
+def upload_consultation(file, user):
+    uploaded_json = file.read()
+
+    consultation_data = json.loads(uploaded_json)
+    consultation_with_responses = ConsultationWithResponses(**consultation_data)
+
+    # convert to JSON and back to get a big dict
+    attrs = json.loads(consultation_with_responses.json())
+
+    responses = attrs["consultation_responses"]
+    consultation_attrs = attrs["consultation"]
+    sections = consultation_attrs.pop("sections")
+
+    consultation_attrs["slug"] = slugify(consultation_attrs["name"])
```

Comment:
How do we enforce uniqueness of the slug (which we need for the URLs to work, and there is also a uniqueness constraint in the models)?

Also, if we are calculating the slug from the name - could we do this in the `save` method in the model?

Reply:
Same for slugs in `Section` and `Question`.

Reply:
I think there's still a uniqueness issue with consultation slug - I can upload 2 consultations with the same name then get an issue trying to to navigate to `consultations/<consultation-slug>`.